.github/workflows/build-and-run.yml

```
@@ -0,0 +1,109 @@
+name: Build-And-Run Tests
+
+on: [push]
```

Can you add `pull_request` here as well?

Ideally, the CI build should be triggered whenever a PR is first created and whenever the PR is updated (by the PR author pushing changes to the PR branch). Right now, a CI build will only be triggered by the latter. (It didn't trigger when you created this PR, for example.)

```suggestion
on: [pull_request, push]
```